kotsadm/pkg/store/s3pg/cluster_store.go

```diff
@@ -39,3 +45,73 @@ func (s S3PGStore) GetClusterIDFromDeployToken(deployToken string) (string, erro
 
 	return clusterID, nil
 }
+
+func (s S3PGStore) LookupClusterID(clusterType string, title string, token string) (string, error) {
+	db := persistence.MustGetPGSession()
+	query := `select id from cluster where cluster_type = $1 and title = $2 and token = $3`
+	row := db.QueryRow(query, clusterType, title, token)
+
+	var clusterID string
+	if err := row.Scan(&clusterID); err != nil {
+		return "", errors.Wrap(err, "failed to scan")
+	}
+
+	return clusterID, nil
+}
+
+func (s S3PGStore) CreateNewShipCluster(userID string, isAllUsers bool, title string, token string) (string, error) {
```

Comment:
Can this just be called `CreateCluster`  Or else we'll forever have to explain the ship thing